### PR TITLE
Redeem intent declared handler

### LINKED ISSUE #174: Redeem intent declared handler
- It updates status to Declared in messages table when RedeemIntentDeclared event is received.
- It should not update status in messages table when message status is not UnDeclared

### PR DESCRIPTION
**Pull request by abhayks1**
- It updates status to Declared in messages table when RedeemIntentDeclared event is received.
- It should not update status in messages table when message status is not Undeclared

Fixes #174

### CODE REVIEW

File: test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import RedeemIntentDeclaredHandler from '../../../../src/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler';
+import Message from '../../../../src/models/Message';
+import {
+  MessageDirection, MessageRepository, MessageStatus, MessageType,
+} from '../../../../src/repositories/MessageRepository';
+import assert from '../../../test_utils/assert';
+import SpyAssert from '../../../test_utils/SpyAssert';
+
+describe('RedeemIntentDeclaredHandler.persist()', (): void => {
+  const transactions = [{
```
Comment (sarvesh-ost) [inline test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts]: Can we have more than 1 transaction here?

File: test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    _messageHash: Web3Utils.keccak256('1'),
+    _redeemer: '0x0000000000000000000000000000000000000001',
+    _redeemerNonce: '1',
+    _beneficiary: '0x0000000000000000000000000000000000000002',
+    _amount: '100',
+    contractAddress: '0x0000000000000000000000000000000000000002',
+    blockNumber: '10',
+  }];
+
+  it('should change message state to source declared', async (): Promise<void> => {
```
Comment (sarvesh-ost) [inline test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts]: One more possible test `should not change message state if source status is already declared`

File: src/handlers/stake_and_mint/MintProgressHandler.ts [hunk 3]
```diff
@@ -15,13 +15,13 @@
 // ----------------------------------------------------------------------------
 
 
-import Logger from '../Logger';
-import Message from '../models/Message';
+import Logger from '../../Logger';
```
[the remark this answers is not in the document]
Comment (abhayks1) [reply src/handlers/stake_and_mint/MintProgressHandler.ts]: Done!

File: src/handlers/stake_and_mint/ProveGatewayHandler.ts [hunk 4]
```diff
@@ -1,11 +1,11 @@
 import assert from 'assert';
 import BigNumber from 'bignumber.js';
 
-import Logger from '../Logger';
-import Gateway from '../models/Gateway';
-import GatewayRepository from '../repositories/GatewayRepository';
-import ContractEntityHandler from './ContractEntityHandler';
-import Utils from '../Utils';
+import Logger from '../../Logger';
```
[the remark this answers is not in the document]
Comment (abhayks1) [reply src/handlers/stake_and_mint/ProveGatewayHandler.ts]: Done!

File: src/handlers/stake_and_mint/StakeProgressHandler.ts [hunk 5]
```diff
@@ -16,13 +16,13 @@
 
 import BigNumber from 'bignumber.js';
 
-import Logger from '../Logger';
-import Message from '../models/Message';
+import Logger from '../../Logger';
```
[the remark this answers is not in the document]
Comment (abhayks1) [reply src/handlers/stake_and_mint/StakeProgressHandler.ts]: Done!

File: src/handlers/stake_and_mint/StakeIntentConfirmHandler.ts [hunk 6]
```diff
@@ -16,13 +16,13 @@
 
 import BigNumber from 'bignumber.js';
 
-import Logger from '../Logger';
-import Message from '../models/Message';
+import Logger from '../../Logger';
```
[the remark this answers is not in the document]
Comment (abhayks1) [reply src/handlers/stake_and_mint/StakeIntentConfirmHandler.ts]: Done!

File: src/handlers/stake_and_mint/StakeRequestHandler.ts [hunk 7]
```diff
@@ -16,11 +16,11 @@
 
 import BigNumber from 'bignumber.js';
 
-import Logger from '../Logger';
-import StakeRequest from '../models/StakeRequest';
-import StakeRequestRepository from '../repositories/StakeRequestRepository';
-import ContractEntityHandler from './ContractEntityHandler';
-import Utils from '../Utils';
+import Logger from '../../Logger';
```
[the remark this answers is not in the document]
Comment (abhayks1) [reply src/handlers/stake_and_mint/StakeRequestHandler.ts]: Done!

File: src/handlers/stake_and_mint/StakeRequestHandler.ts [hunk 8]
```diff
@@ -89,25 +89,24 @@ export default class StakeRequestHandler extends ContractEntityHandler<StakeRequ
           const blockNumber = new BigNumber(transaction.blockNumber);
 
           const stakeRequest = await this.stakeRequestRepository.get(stakeRequestHash);
-          if (stakeRequest && blockNumber.gt(stakeRequest.blockNumber!)) {
+          if (stakeRequest && blockNumber.gt(stakeRequest.blockNumber)) {
```
[the remark this answers is not in the document]
Comment (abhayks1) [reply src/handlers/stake_and_mint/StakeRequestHandler.ts]: added documentation.

File: test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts (hunk 2 above)
Comment (abhayks1) [reply test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts]: The test is already there. See [line no 57](https://github.com/mosaicdao/facilitator/blob/a25ca891b3aeab57e43db13a18bc1c3dc4af6fe7/test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts#L57)

File: test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts (hunk 1 above)
Comment (abhayks1) [reply test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts]: With same messageHash right?

**Comment by abhayks1** [comment]
Available for review 🎸

File: test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts (hunk 2 above)
Comment (abhayks1) [reply test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts]: Added test.

File: src/handlers/stake_and_mint/MintProgressHandler.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (sarvesh-ost) [reply src/handlers/stake_and_mint/MintProgressHandler.ts]: Please update test folder name as-well.

File: test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts (hunk 1 above)
Comment (sarvesh-ost) [reply test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts]: no different message hash.

File: src/handlers/stake_and_mint/MintProgressHandler.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (abhayks1) [reply src/handlers/stake_and_mint/MintProgressHandler.ts]: Done!

File: test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts (hunk 1 above)
Comment (abhayks1) [reply test/handlers/redeem_and_unstake/RedeemIntentDeclaredHandler/persist.test.ts]: Done!

**Pull request merged**